packages/consts/src/regexs.ts

```diff
@@ -1,14 +1,22 @@
 /**
  * Email validation regexp adapted from https://html.spec.whatwg.org/multipage/forms.html#valid-e-mail-address
  * with our restriction that hostname must be a TLD! (will not match example@localhost)
+ * and two consecutive dots in name are not allowed (based on Mailgun convention, will not match ex..amle@example.com)
  */
-export const EMAIL_REGEX_STR = '[a-zA-Z0-9.!#$%&\'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+'; // eslint-disable-line max-len
 
-/**
- * Matches a string containing valid email
- * Hostname must be a TLD! (will not match example@localhost)
- */
-export const EMAIL_REGEX = new RegExp(`^${EMAIL_REGEX_STR}$`);
+// Parts for building an email regex (email will be constructed as `name@domain`)
+// name parts can be alnum + some special characters
+const namePartSubRegexStr = '[a-zA-Z0-9!#$%&\'*+/=?^_`{|}~-]+';
+// name is 1+ name parts joined by periods (no leading or dangling period, no consecutive periods)
+const nameSubRegexStr = `${namePartSubRegexStr}(?:\\.${namePartSubRegexStr})*`;
+// domain parts can be alnum and dash characters (no leading and dangling dashes, max 63 chars long)
+const domainPartSubRegexStr = '[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?'; //
+// doman is 2+ domain parts joined by periods (no leading or dangling period, no consecutive periods)
+const domainSubRegexStr = `${domainPartSubRegexStr}(?:\\.${domainPartSubRegexStr})+`;
+
+export const EMAIL_REGEX_STR = `^${nameSubRegexStr}@${domainSubRegexStr}$`;
+
+export const EMAIL_REGEX = new RegExp(EMAIL_REGEX_STR);
```

Comment:
Also, can you add back the JSDocs just before the `export const XXX` lines, so that they're associated with the constants, and show in editor hints etc.?

Reply:
thanks, done